### PR TITLE
Proposal to integrate ifttt/maker actions

### PR DESCRIPTION
**Pull request by chof747**
with this pull request I propose to integrate sending web messages to to ifttt/maker channel which can be used to integrate events in pilight with the if-this-than-that web service (https://ifttt).

Attached is also a documentation of the new action - ready for integration in wiki: https://docs.google.com/document/d/17tKzVZzSqZr5w6VL2WKUgAuoOeEbzNmZOgxVrnHnmq0/pub

The pull request has the following content:

- maker.c and maker.h file containing the new action
- change to http.c and http.h: added a function to urlencode strings (needed for parameter encoding)

Please consider if you want to integrate it into pilight.

Kind regards

Christian

### CODE REVIEW

File: libs/pilight/events/actions/maker.c [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+	      }
+	    }
+
+      if(nrvalues != 1) {
+        logprintf(LOG_ERR,
+                  "maker web request action \"%s\" only takes one argument",
+                  MAKER_VALUE_PARAMS[i]);
+        return -1;
+	    }
+  	}
```
Comment (CurlyMoo) [inline libs/pilight/events/actions/maker.c]: There is something wrong with this accolade.

**Comment by CurlyMoo** [comment]
1. Please leave the .gitignore
2. A urldecode already exists in pilight
3. Can you adapt the pull request according to the contribution guide?
4. Can you change the `maker` to something more verbose?

**Comment by chof747** [comment]
Ok I will change the pull request as you recommend. Would iftttmaker be a proper name?

**Comment by chof747** [comment]
I close this pull request corrections will be posted in new one as it was done from the wrong branch.

**Pull request closed without merging**